lib/pagination/adapter.js

```diff
@@ -1,50 +1,56 @@
-const { invalidPageNumber, invalidLimitNumber } = require('./errors');
+const { invalidPageNumberError, invalidLimitNumberError } = require('./errors');
 
-let content = undefined;
-let page = undefined;
-let limit = undefined;
-let request = undefined;
+let adapterParams = { content: undefined, page: undefined, limit: undefined, request: undefined };
 
-const createAdapter = (data, currentPage, limitParam, req) => {
-  if (typeof currentPage !== 'number' || currentPage <= 0) throw invalidPageNumber('Invalid Page Number');
-  if (typeof limitParam !== 'number' || limitParam <= 0) throw invalidLimitNumber('Invalid Limit Number');
+const setAdapterParams = params => ({ ...adapterParams } = { ...params });
```

Comment:
do you need to return `adapterParams`?
You can do `const setAdapterParams = params => { adapterParams = { ...params }; };`